### PR TITLE
docs(tutorial): add Kris suggested intro

### PR DESCRIPTION
**Pull request by jrhender**

### CODE REVIEW

File: apps/nestjs-wallet/docs/tutorial.md [hunk 1]
```diff
@@ -1,5 +1,56 @@
...
+        "familyName": "schema:familyName",
+        "gender": "schema:gender",
+        "givenName": "schema:givenName",
+        "lprCategory": "ctzn:lprCategory",
+        "lprNumber": "ctzn:lprNumber",
+        "residentSince": {"@id": "ctzn:residentSince", "@type": "xsd:dateTime"}
+        ...
+
+To get the definition of the field "gender" which points to the definition "schema:gender", use the URI of "schema" which points to "http://schema.org/", so we will have a definition available at "http://schema.org/gender".
+TODO interpretation of id and type.
```
Comment (jrhender) [inline apps/nestjs-wallet/docs/tutorial.md:29]: I'm not sure what you mean by "interpretation of id and type". 
I think probably you are looking for an explanation of why these two lines are in the `context`:
```
"id": "@id",
"type": "@type",
```
If so, the short answer is that is they are aliasing `@id` and `@type` to `id` and `type` so that they are compatible with `json`. (At least I think that's the answer 😄 )

File: apps/nestjs-wallet/docs/tutorial.md [hunk 2]
```diff
@@ -1,5 +1,56 @@
 # Tutorial
 
+## Business overview
+The business objective of this tutorial is to demonstrate how to issue a proof in the form of a Verifiable Credential (VC) to a known user connected to a portal that we will call "authority portal".
```
Comment (jrhender) [inline apps/nestjs-wallet/docs/tutorial.md:4]: I'm not sure what you mean by "a proof in the form of a Verifiable Credential". A Verifiable Credential contains a proof but isn't really a "form" of a proof, in my opinion.
Perhaps we can simplify it as shown below:
```suggestion
The business objective of this tutorial is to demonstrate how to issue a Verifiable Credential (VC) to a known user connected to a portal that we will call "authority portal".
```

File: apps/nestjs-wallet/docs/tutorial.md [hunk 3]
```diff
@@ -1,5 +1,56 @@
...
+
+Business workflow:
+- The citizen requests a "PermanentResidentCard" proof on the autority portal
+- The autority portal displays a QR code for the citizen to scan with his mobile wallet.
+- The citizen's mobile wallet contacts the autority portal to prove that he has a DID
+- The citizen's mobile wallet contacts the autority portal again to receive the "PermanentResidentCard"
+- The autority portal responds by sending the "PermanentResidentCard" proof containing the citizen's information signed by the autority portal.
+
+## Technical overview
+From a technical point of view, in this tutorial, we have access to the server API but no mobile wallet is available. So we will use the server API for the portal and the mobile wallet.
```
Comment (jrhender) [inline apps/nestjs-wallet/docs/tutorial.md:41]: I'm a bit confused by this line. It says that there is "no mobile wallet available" but then says that "we will use the server API for the portal and the mobile wallet". How can we use the mobile wallet if there is none available?

File: apps/nestjs-wallet/docs/tutorial.md [hunk 4]
```diff
@@ -1,5 +1,56 @@
...
+
+We will follow the following steps in this tutorial:
+- [Authority portal action] Create a DID for the autority portal in the form of a key DID method
+- [Authority portal action] Register the DID as default DID for the authority portal
+- [Authority portal action] List the available proofs that the autority portal can issue, an other tutorial is dedicated to demonstrate how to add new available proof for the authority portal
+- [Authority portal action] Create a VC request url that can be transmitted to the citizen as a QR code or a deep link
```
Comment (jrhender) [inline apps/nestjs-wallet/docs/tutorial.md:47]: These are the same action I believe. Currently `GET Credential Offer` returns both the available Credential types and the url to start the workflow to request the credentials.
```
{
    "typeAvailable": "PermanentResidentCard",
    "vcRequestUrl": "http://localhost:3000/elia-issuer/start-workflow/permanent-resident-card"
}
```
My suggestion would be to modify as shown.
```suggestion
- [Authority portal action] Create a "credential offer" which lists the available credential types that the autority portal can issue and a VC request url that be used to start a request flow. This credential offer can be transmitted to the citizen as a QR code or a deep link.
```

File: apps/nestjs-wallet/docs/tutorial.md [hunk 5]
```diff
@@ -1,5 +1,56 @@
...
+- [Citizen action] Create a DID for the citizen
+- [Citizen action] The citizen authenticates himself on the autority portal with his DID and the autority portal responds with a Verifiable Presentation that proves the successful connection of the user
+- [Citizen action] The citizen, using the unique endpoint received in the QR code (VC request URL), will transmit the proof of connection in the form of the VP received in the previous step. The autority portal will create a VC using the available data of the citizen in its database, sign the VC and transmit it as a response to the request.
```
Comment (jrhender) [inline apps/nestjs-wallet/docs/tutorial.md:50]: I think we need to edit this section a bit. I don't think the citizen is authenticating himself on the authority portal with his DID. I think the citizen is doing two things in a real-world flow:
- Authenticating himself as his real world identity (which isn't really shown/captured in the flow at the moment)
- Proving control of a DID
With these two pieces of info, the authority can issue a VC to the DID for the citizen's permanent resident information

File: apps/nestjs-wallet/docs/tutorial.md [hunk 6]
```diff
@@ -1,5 +1,56 @@
...
+
+TODO: tutorial on how to create new proofs available to the server
```
Comment (jrhender) [inline apps/nestjs-wallet/docs/tutorial.md:52]: By "proofs", do you mean credential types?

**Review by jrhender — COMMENTED** [review]
@gixita this section is a great addition. I think it just needs some minor edits. Let's discuss further when we have a chance.

File: apps/nestjs-wallet/docs/tutorial.md (hunk 1 above)
Comment (gixita) [reply apps/nestjs-wallet/docs/tutorial.md:29]: Ho do you know that type can take one of the https://w3id.org/citizenship/v1 values? If PermanentResident and Person where at a lower level than PermanentResidentCard, that could be understandable.

File: apps/nestjs-wallet/docs/tutorial.md (hunk 2 above)
Comment (gixita) [reply apps/nestjs-wallet/docs/tutorial.md:4]: Ok, perfect

File: apps/nestjs-wallet/docs/tutorial.md (hunk 3 above)
Comment (gixita) [reply apps/nestjs-wallet/docs/tutorial.md:41]: I mean ; we use the server API also in place of the mobile wallet

File: apps/nestjs-wallet/docs/tutorial.md (hunk 4 above)
Comment (gixita) [reply apps/nestjs-wallet/docs/tutorial.md:47]: But when you get the vcRequestUrl, you need to make a new action to get the : 
`{
    "errors": [],
    "vpRequest": {
        "challenge": "57ca126c-acbf-4da4-8f79-447150e93128",
        "query": [
            {
                "type": "DIDAuth"
            }
        ],
        "interact": {
            "service": [
                {
                    "type": "VcApiPresentationService2021",
                    "serviceEndpoint": "http://localhost:3000/elia-issuer/active-flows/c4e0e642-02a2-4a1a-91c6-40069b328238"
                }
            ]
        }
    }
}`
This will be transformed in a qr code.
So it seems to be two actions to the server api, one to know what is available and the second to get the content of the qr code

File: apps/nestjs-wallet/docs/tutorial.md (hunk 5 above)
Comment (gixita) [reply apps/nestjs-wallet/docs/tutorial.md:50]: Proving control of a DID was call authentication in the tutorial, but we can change it.
Your title is : Create a DID authentication

File: apps/nestjs-wallet/docs/tutorial.md (hunk 1 above)
Comment (jrhender) [reply apps/nestjs-wallet/docs/tutorial.md:29]: Hmm, well I think `type` is a generic concept from JSON-LD (https://json-ld.org/spec/latest/json-ld/#specifying-the-type) and the VC data model seems to allow any URI for the type. https://www.w3.org/TR/vc-data-model/#types
So I'm not sure there's a way to unambiguously classify a URI as being suitable to be a VC type.
Does that answer your question?

File: apps/nestjs-wallet/docs/tutorial.md (hunk 3 above)
Comment (jrhender) [reply apps/nestjs-wallet/docs/tutorial.md:41]: Ah, I see now. What do you think of this suggestion?
```suggestion
From a technical point of view, in this tutorial, we have access to the server API but no mobile wallet is available. So we will use the server API for both roles of the portal and the mobile wallet.
```

File: apps/nestjs-wallet/docs/tutorial.md (hunk 4 above)
Comment (jrhender) [reply apps/nestjs-wallet/docs/tutorial.md:47]: Ah gotcha. Yes, I'm not sure whether or not the "mobile wallet" would consume the `Credential Offer` or the `VP Request`. I suppose the `Credential Offer` isn't standardized, so you're probably right that it is better to assume that the mobile wallet would consume the `VP Request` as a QR code. How about this wording?
```suggestion
- [Authority portal action] List the available credentials that the autority portal can issue. The citizen, using the portal, would select a credential that they want to request.
- [Authority portal action] Create a [VP Request](https://w3c-ccg.github.io/vp-request-spec/) that can be transmitted to the citizen as a QR code or a deep link
```

File: apps/nestjs-wallet/docs/tutorial.md (hunk 5 above)
Comment (jrhender) [reply apps/nestjs-wallet/docs/tutorial.md:50]: Hmm, yes, I should probably have said "authenticating as a DID" instead of "proving control of a DID".

However, I think this section still needs some edits. For instance, I don't think that this section is correct:
> The citizen authenticates himself on the autority portal with his DID and the autority portal responds with a Verifiable Presentation that proves the successful connection of the user

We can discuss further during a call and/or I can suggest some edits tomorrow.

File: apps/nestjs-wallet/docs/tutorial.md (hunk 6 above)
Comment (jrhender) [reply apps/nestjs-wallet/docs/tutorial.md:52]: Ok, sounds good. As the functionality to add new credential issuance types is tracked by this issue https://github.com/energywebfoundation/ssi/issues/13 , perhaps we can remove this TODO for now? We can also keep it if you prefer.
```suggestion
```

**Review by jrhender — COMMENTED** [review]
@gixita , I replied to your comments. Please take a look when you have a moment 😄

File: apps/nestjs-wallet/docs/tutorial.md (hunk 1 above)
Comment (jrhender) [reply apps/nestjs-wallet/docs/tutorial.md:29]:
```suggestion
Regarding the aliasing of `id` to `@id` and `type` to `@type`, see the [VC specification](https://www.w3.org/TR/vc-data-model/#syntactic-sugar).
```
